packages/core/react/src/WalletProvider.tsx

```diff
@@ -109,12 +110,10 @@ export function WalletProvider({
         if (autoConnect !== true || !adapter) {
             return;
         }
-        if (walletName === SolanaMobileWalletAdapterWalletName && getIsMobile(adaptersWithStandardAdapters)) {
-            return (adapter as SolanaMobileWalletAdapter).autoConnect_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.bind(adapter);
-        } else {
-            return adapter.connect.bind(adapter);
-        }
-    }, [adapter, adaptersWithStandardAdapters, autoConnect, walletName]);
+
+        // Not sure why this is needed, adapter is already WalletAdapter<string> | SolanaMobileWalletAdapter
+        return (adapter as WalletAdapter<string> | SolanaMobileWalletAdapter).autoConnect.bind(adapter);
```

Comment:
Yeah, it does seem like this can just be `adapter.autoConnect.bind(adapter)` now, right?